### PR TITLE
chore: add doc comments to the capabilities

### PR DESCRIPTION
**Pull request by Gozala**
## overview

- Adds bunch of comments to the capabilities
- Makes mandatory fields non-optional (I had to fix bugs in ucanto and update to new version to fix this)

### CODE REVIEW

File: packages/access/src/capabilities/upload.js [hunk 1]
```diff
@@ -46,15 +82,30 @@ export const add = base.derive({
...
+  /**
+   * `upload/add` can be derived from the `upload/*` & `*` capability
+   * as long as `with` fields match.
+   */
   derives: equalWith,
 })
 
+/**
+ * Capability removes an upload (identified by it's root CID) from the upload
+ * list. Please not that removing an upload does not delete corresponding shards
```
Comment (alanshaw) [inline packages/access/src/capabilities/upload.js:94]:
```suggestion
 * list. Please note that removing an upload does not delete corresponding shards
```

File: packages/access/src/capabilities/voucher.js [hunk 2]
```diff
@@ -16,25 +37,47 @@ export const voucher = any.derive({
 
 const base = any.or(voucher)
 
+/**
+ * Capability can be invoked by an agent to claim a voucher for a specific
+ * user identifier (currently email address).
+ *
+ * The agent MAY issue claim with own DID or a DID it is delegate of. If `with`
+ * is different from `iss`, it is implied that it voucher is claimed for the
```
Comment (alanshaw) [inline packages/access/src/capabilities/voucher.js:45]:
```suggestion
 * is different from `iss`, it is implied that the voucher is claimed for the
```

File: packages/access/src/capabilities/voucher.js [hunk 3]
```diff
@@ -16,25 +37,47 @@ export const voucher = any.derive({
...
+ * DID in the `with` field. If `with` is same as `iss` it is implies that
+ * voucher is claim for unspecified `did`.
```
Comment (alanshaw) [inline packages/access/src/capabilities/voucher.js:47]:
```suggestion
 * voucher is claimed for an unspecified `did`.
```

File: packages/access/src/capabilities/voucher.js [hunk 4]
```diff
@@ -16,25 +37,47 @@ export const voucher = any.derive({
...
+       */
+      product: Product,
+      /**
+       * Verifiable identity on who's behalf behalf claim is made.
+       */
+      identity: Identity,
+      /**
+       * Optional service DID who's voucher is been requested.
+       */
+      service: URI.match({ protocol: 'did:' }).optional(),
```
Comment (alanshaw) [inline packages/access/src/capabilities/voucher.js:65]: Can we not do this?

```suggestion
      service: Service.optional(),
```

File: packages/access/src/capabilities/voucher.js [hunk 5]
```diff
@@ -43,19 +86,36 @@ export const redeem = voucher.derive({
...
+       */
+      product: Product,
+      /**
+       * Verifiable identity to whom voucher is issued. It is a `mailto:` URL
+       * where this delegation is typically sent.
+       */
+      identity: Identity,
+      /**
+       * Space identifier where voucher can be redeemed. When service delegates
+       * `vourche/redeem` to the user agent it may omit this field to allow
```
Comment (alanshaw) [inline packages/access/src/capabilities/voucher.js:101]:
```suggestion
       * `voucher/redeem` to the user agent it may omit this field to allow
```

**Review by alanshaw — APPROVED** [review]
😍

File: packages/access/src/capabilities/store.js [hunk 6]
```diff
@@ -23,17 +33,41 @@ export const store = any.derive({
...
+      size: Schema.integer(),
+      /**
+       * Agent may optionally provide a link to a related CAR file using `origin`
+       * field. This is useful when storing large DAGs, agent could shard it
+       * across multiple CAR files and then link each shard with a previous one.
+       *
+       * Providing this relation tells service that given CAR is shard of the
+       * larger DAG as opposed to it being intentionally partial DAG.
+       */
       origin: Link.optional(),
```
[the remark this answers is not in the document]
Comment (Gozala) [reply packages/access/src/capabilities/store.js:70]: Just omit it, should I add a comment

File: packages/access/src/capabilities/store.js [hunk 7]
```diff
@@ -23,17 +33,41 @@ export const store = any.derive({
...
+       * for this exact size. Attempt to write a larger CAR file will fail.
+       */
+      size: Schema.integer(),
+      /**
+       * Agent may optionally provide a link to a related CAR file using `origin`
+       * field. This is useful when storing large DAGs, agent could shard it
+       * across multiple CAR files and then link each shard with a previous one.
+       *
+       * Providing this relation tells service that given CAR is shard of the
+       * larger DAG as opposed to it being intentionally partial DAG.
```
Comment (Gozala) [inline packages/access/src/capabilities/store.js:68]:
```suggestion
       * larger DAG as opposed to it being intentionally partial DAG. When DAG is not
       * sharded, there will be only one `store/add` with `origin` left out.
```

File: packages/access/src/capabilities/store.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (Gozala) [reply packages/access/src/capabilities/store.js:71]: added a comment to clarify this

**Pull request merged**